src/openrct2/platform/Platform.Win32.cpp

```diff
@@ -90,6 +91,105 @@ namespace Platform
         return String::ToUtf8(result);
     }
 
+    std::string GetOsName()
+    {
+        auto hModule = GetModuleHandleA("ntdll.dll");
+        RTL_OSVERSIONINFOW rovi = {};
+        if (hModule != nullptr)
+        {
+            using RtlGetVersionPtr = long(WINAPI*)(PRTL_OSVERSIONINFOW);
+#    if defined(__GNUC__) && __GNUC__ >= 8
+#        pragma GCC diagnostic push
+#        pragma GCC diagnostic ignored "-Wcast-function-type"
+#    endif
+            auto fn = reinterpret_cast<RtlGetVersionPtr>(GetProcAddress(hModule, "RtlGetVersion"));
+#    if defined(__GNUC__) && __GNUC__ >= 8
+#        pragma GCC diagnostic pop
+#    endif
+            if (fn != nullptr)
+            {
+                rovi.dwOSVersionInfoSize = sizeof(rovi);
+                if (fn(&rovi) != 0)
```

Comment:
`RtlGetVersion` can only return `STATUS_SUCCESS`, so this block will never be hit.

Reply:
Removed the redundant code